jobQueue/setup.js

```diff
@@ -18,19 +18,46 @@ const conf = require('../config');
 const featureToggles = require('feature-toggles');
 const urlParser = require('url');
 const kue = require('kue');
+const activityLogUtil = require('../utils/activityLog');
+
 const redisOptions = {
   redis: conf.redis.instanceUrl.queue,
 };
+
+const express = require('express');
+const server = require('http').Server(express());
+const SHUTDOWN_TIMEOUT = 5000;
+const SUCCESSFUL_EXIT = 0;
+
 const redisInfo = urlParser.parse(redisOptions.redis, true);
 if (redisInfo.protocol !== PROTOCOL_PREFIX) {
   redisOptions.redis = 'redis:' + redisOptions.redis;
 }
 
 const jobQueue = kue.createQueue(redisOptions);
 
+process.on('SIGTERM', () => {
+  const start = Date.now();
+  server.close(() => { // Stop http server and stop accepting any new request.
+    jobQueue.shutdown(SHUTDOWN_TIMEOUT, (err) => {
```

Comment:
I think the callback fn which we want to invoke on shutdown should be defined and exported from jobQueue setup, but the process.on and server.close code should be up a level, not inside jobqueue

Reply:
OK, got that. Do you think index.js would be a better place?